tools/dump_with_cuda_graph.py

```
@@ -0,0 +1,241 @@
+
+    # Special-case `python ...` so we run inside this interpreter (which
+    # already has the monkey-patch active). For any other executable, fall
+    # back to os.execvp; that re-execs into a fresh interpreter, so the
+    # monkey-patch must be re-installed there. Document this.
+    head = target_argv[0]
+    if head in ("python", "python3", sys.executable):
+        # Run as a Python script / module in-process so our patch survives.
+        rest = target_argv[1:]
+        if rest and rest[0] == "-m":
+            module = rest[1]
+            sys.argv = [module] + rest[2:]
```

Ruff (0.15.11)

[warning] Consider `[module, *rest[2:]]` instead of concatenation

Replace with `[module, *rest[2:]]`

(RUF005)